openedx/core/djangoapps/user_authn/views/password_reset.py

```diff
@@ -659,3 +662,54 @@ def password_reset_token_validate(request):
         AUDIT_LOG.exception("Invalid password reset confirm token")
 
     return JsonResponse({'is_valid': is_valid})
+
+
+def _is_token_available(**kwargs):
+    """
+    Helper function to test that token
+    is available in view reset url
+    """
+    uidb36 = kwargs.get('uidb36')
+    token = kwargs.get('token')
+
+    if not uidb36 or not token:
+        return False
+    try:
+        _ = base36_to_int(uidb36)
+    except ValueError:
+        return False
+    return True
+
+
+@require_POST
+@ensure_csrf_cookie
+def password_reset_logistration(request, **kwargs):
+    """Reset learner password using passed token and new credentials"""
+
+    reset_status = False
+    uidb36 = kwargs.get('uidb36')
+    token = kwargs.get('token')
+
+    if not _is_token_available(**kwargs):
```

Comment:
Can we leverage `default_token_generator.check_token()` here instead of this custom helper function?

Reply:
I have done some of the optimization moving token check above password validation. Secondly default_token_generator.check_token() checks only the token part not the whole string. So still we need to validate kwargs.get('uidb36') properly.